server/controllers/establishmentRollController.ts

```diff
@@ -108,15 +131,31 @@ export default class EstablishmentRollController {
       const { livingUnitId } = req.params
       const { clientToken } = req.middleware
 
-      const [prisonersCurrentlyOut, location] = await Promise.all([
-        this.movementsService.getOffendersCurrentlyOutOfLivingUnit(clientToken, livingUnitId),
-        this.locationService.getLocationInfo(clientToken, livingUnitId),
-      ])
+      const useLocationsApi = this.isResidentialLocationsEnabledForThisPrison(
+        await this.serviceData.getServiceData(res),
+      )
 
-      res.render('pages/currentlyOut', {
-        prisoners: prisonersCurrentlyOut,
-        location,
-      })
+      if (useLocationsApi) {
+        const [prisonersCurrentlyOut, location] = await Promise.all([
+          this.movementsService.getOffendersCurrentlyOutOfBed(clientToken, livingUnitId),
+          this.locationService.getInternalLocationInfo(clientToken, livingUnitId),
+        ])
+
+        res.render('pages/currentlyOut', {
+          prisoners: prisonersCurrentlyOut,
+          locationName: location.localName ? location.localName : location.pathHierarchy,
+        })
+      } else {
+        const [prisonersCurrentlyOut, location] = await Promise.all([
+          this.movementsService.getOffendersCurrentlyOutOfLivingUnit(clientToken, livingUnitId),
+          this.locationService.getLocationInfo(clientToken, livingUnitId),
+        ])
+
+        res.render('pages/currentlyOut', {
+          prisoners: prisonersCurrentlyOut,
+          locationName: location.userDescription ? location.userDescription : location.description,
+        })
+      }
```

Comment:
Would rather return the res.render from the if block and get rid of the else

Reply:
I'll do that in next PR